### ISSUE TITLE
Cancel Run Action without Run Id throws Forbidden Error

### ISSUE THREAD

**Issue by RodrigoCuryZup** [stack-spot/runtime-cancel-run-action#3]
## Context

An Itaú Client has shown us the following error in his pipeline:
```
Deploying Self-Hosted...
- Error starting self hosted deploy run
- Status: 409
- Error: Conflict
- Response: {"code":"RTM_DEPLOYMENT_INFRA_ALREADY_RUNNING_CONFLICT_ERROR","status":409,"details":"There is already a deployment (that modifies this infra) running","validationDetails":null}
Error: Process completed with exit code 1.

e  erro no cancel

Authenticating..
Cancelling Run...
- Error cancelling run
- Status: 403
- Error: Forbidden
- Response: 
Error: Process completed with exit code 1.
```

The issue above is that the Cancel Run Action does not have a Run Id since the deploy has not been created.

<details><summary>This is OPA's Log for the validation.</summary>
<p>
<code>{"bundles":{"default":{}},"decision_id":"f9d00b2a-9647-4ece-8b60-c820e01e1565","input":{"attributes":{"destination":{"address":{"socketAddress":{"address":"10.144.58.86","portValue":8000}}},"metadataContext":{},"request":{"http":{"headers":{":authority":"runtime-manager.v1.stackspot.com",":method":"POST",":path":"/v1/run/cancel/",":scheme":"https","accept":"*/*","accept-encoding":"gzip, deflate","authorization":"Bearer ....","content-length":"0","content-type":"application/json","user-agent":"python-requests/2.31.0","via":"1.1 e0907aa05825aa9eeecded292d438c6e.cloudfront.net (CloudFront)","x-amz-cf-id":"S3RmjFFVkVltpp-lVhoR_RoPEp-RdnT7mOSrlZw6seErTu_AZ-bg2A==","x-amzn-trace-id":"Root=1-65b0f7da-71dcacf06c3d037149966ca4","x-forwarded-for":"54.232.98.19","x-forwarded-host":"runtime-manager.v1.stackspot.com","x-forwarded-port":"443","x-forwarded-proto":"https","x-forwarded-scheme":"https","x-origin-verify":"rODs7RJxZ8uIky8hMqB0rxPoIxvz476OkrShfGU5OF8O36zFTLCmVIUNEtrp3geg","x-original-forwarded-for":"54.232.98.19, 64.252.69.109","x-real-ip":"54.232.98.19","x-request-id":"a323ad4ad5b8e99941d60abd0d71e4b6","x-scheme":"https"},"host":"runtime-manager.v1.stackspot.com","id":"8766460387303765325","method":"POST","path":"/v1/run/cancel/","protocol":"HTTP/1.1","scheme":"https"},"time":"2024-01-24T11:43:22.323642Z"},"source":{"address":{"socketAddress":{"address":"10.144.31.146","portValue":52984}}}},"parsed_body":null,"parsed_path":["v1","run","cancel",""],"parsed_query":{},"truncated_body":false,"version":{"encoding":"protojson","ext_authz":"v3"}},"labels":{"id":"2c7d1358-8b53-4ae3-9643-ee5f9bbca73e","version":"0.45.0-envoy-10"},"level":"info","metrics":{"counter_rego_builtin_http_send_interquery_cache_hits":1,"timer_rego_builtin_http_send_ns":72738690,"timer_rego_query_eval_ns":77664906,"timer_server_handler_ns":77946380},"msg":"Decision Log","path":"com/stackspot/authz/allow","result":false,"time":"2024-01-24T11:43:22Z","timestamp":"2024-01-24T11:43:22.40219196Z","type":"openpolicyagent.org/decision_logs"}</code>

</p>
</details> 

The important parts from the logs are the following:
- `"method":"POST","path":"/v1/run/cancel/"`
- `"result":false`

Since it has no run_id OPA does not recognize the URI
```
is_path_run_cancel_run if path_match(["v1","run","cancel",ULID_REGEX])
```

## What happened?

The Action called does not validate **_RUN_ID_** if it is null, throwing an error when the deploy has failed to start and does not create the variable.

## What did you expect to happen?

In Discussion...

## How to reproduce it (as little and accurately as possible)?

- Deploy an application with self-hosted that fails (use a tag that has already been used)
- Use the Cancel Run Action

**Comment by RodrigoCuryZup** [comment]
@GuillaumeFalourd I wonder, when theres no run_id, do you think that the implementation of the cancel action should:
- Action should validate and log that no run_id been passed and `exit(1)`
- Action should validate and log that no run_id been passed and `exit(0)`
- We should update the doc so that the validation becomes `if: ${{ always() && contains(needs.*.result, 'failure') && needs.orchestration.outputs.run_id != '' }}`
- A combination of the above (1 or 2) and 3

**Comment by RodrigoCuryZup** [comment]
Tag v1 of cancel action was succesfully merged by [PR#4](https://github.com/stack-spot/runtime-cancel-run-action/pull/4)
